src/Server/WishList.fs

```diff
@@ -1,49 +1,50 @@
 /// Wish list API web parts and data access functions.
 module ServerCode.WishList
 
-open Suave
-open Suave.Logging
-open Suave.RequestErrors
-open Suave.ServerErrors
+open Giraffe
+open Giraffe.Tasks
+open RequestErrors
+open ServerErrors
 open ServerCode.Domain
-open Suave.Logging.Message
-
-let logger = Log.create "FableSample"
+open Microsoft.AspNetCore.Http
+open Microsoft.Extensions.Logging
+open System.Threading.Tasks
 
 /// Handle the GET on /api/wishlist
-let getWishList getWishListFromDB (ctx: HttpContext) =
-    Auth.useToken ctx (fun token -> async {
+let getWishList (getWishListFromDB: string -> Task<WishList>) next (ctx: HttpContext) =
+    Auth.useToken next ctx (fun token -> task {
         try
             let! wishList = getWishListFromDB token.UserName
-            return! Successful.OK (FableJson.toJson wishList) ctx
+            return! FableJson.serialize wishList next ctx
         with exn ->
-            logger.error (eventX "SERVICE_UNAVAILABLE" >> addExn exn)
-            return! SERVICE_UNAVAILABLE "Database not available" ctx
+            let msg = "Database not available"
+            let logger = ctx.GetLogger "wishlist"
+            logger.LogError (EventId(), exn, msg)
+            return! SERVICE_UNAVAILABLE "Database not available" next ctx
     })
 
 /// Handle the POST on /api/wishlist
-let postWishList saveWishListToDB (ctx: HttpContext) =
-    Auth.useToken ctx (fun token -> async {
+let postWishList (saveWishListToDB: WishList -> Task<unit>) next (ctx: HttpContext) =
+    Auth.useToken next ctx (fun token -> task {
         try
-            let wishList = 
-                ctx.request.rawForm
-                |> System.Text.Encoding.UTF8.GetString
-                |> FableJson.ofJson<Domain.WishList>
+            let! wishList = FableJson.getJsonFromCtx<Domain.WishList> ctx
             
             if token.UserName <> wishList.UserName then
-                return! UNAUTHORIZED (sprintf "WishList is not matching user %s" token.UserName) ctx
+                return! FORBIDDEN (sprintf "WishList is not matching user %s" token.UserName) next ctx
             else
                 if Validation.verifyWishList wishList then
                     do! saveWishListToDB wishList
-                    return! Successful.OK (FableJson.toJson wishList) ctx
+                    return! FableJson.serialize wishList next ctx
                 else
-                    return! BAD_REQUEST "WishList is not valid" ctx
+                    return! BAD_REQUEST "WishList is not valid" next ctx
         with exn ->
-            logger.error (eventX "Database not available" >> addExn exn)
-            return! SERVICE_UNAVAILABLE "Database not available" ctx
+            let msg = "Database not available"
+            let logger = ctx.GetLogger "wishlist"
+            logger.LogError (EventId(), exn, msg)
```

Comment:
Ah, so this is how you use the logger. I haven't activated it yet 😅

Reply:
yes it's latest fix in girafe